javascript/lang/security/audit/sqli/node-postgres-sqli.yaml

```diff
@@ -0,0 +1,63 @@
+rules:
+- id: node-postgres-sqli
+  patterns:
+  - pattern-either:
+    - patterns:
+      - pattern-either:
+        - pattern: |
+            $DB.$METHOD(...,$QUERY,...)
+        - pattern: |
+            $DB.$METHOD(..., $QUERY, ...).$INTFUNC(...)
+      - pattern-either:
+        - pattern-inside: |
+            $QUERY = $X + $Y
+            ...
+        - pattern-inside: |
+            $QUERY += $X
+            ...
+        - pattern-inside: |
+            $QUERY = $X.concat($Y)
+            ...
+      - pattern-not-inside: |
+          $QUERY += "..."
+          ...
+      - pattern-not-inside: |
+          $QUERY = "..." + "..."
+          ...
+      - pattern-not-inside: |
+          $QUERY = $X.concat("...")
+          ...
+    - pattern: $DB.$METHOD(..., $X + $Y, ...)
+    - pattern: $DB.$METHOD(..., $X.concat($Y), ...)
+  - pattern-either:
+    - pattern-inside: |
+        const { $CLIENT } = require('pg')
+        ...
+        $DB = new $CLIENT(...)
+        ...
+    - pattern-inside: |
+        const { $POOL } = require('pg')
+        ...
+        const $NEWPOOL = new $POOL(...)
+        ...
+        $NEWPOOL.connect((..., $DB, ...) => {
+            ...
+        })
+  - pattern-not: $DB.$METHOD(..., "..." + "...", ...)
+  - pattern-not: $DB.$METHOD(..., $X.concat("..."), ...)
+  - metavariable-regex:
+      metavariable: $METHOD
+      regex: ^(query)$
```

Comment:
If `$METHOD` can only be `query` I think we can significantly speed up this rule by specifying the constant `query` everywhere instead of using a metavariable.

Reply:
I second this 👍

Reply:
done!